### DISCUSSION TITLE
LF in Combination with Zoxide and NuShell

### DISCUSSION THREAD

**Discussion by msMatix** [gokcehan/lf#1558] [Q&A] [answered]
Hey :),

I want to use my lf in combination with zoxide. 
My Setup is:
- wezterm 
- nushell
- OS: Windows

For that I implemented the following cmd in my lfrc:
`cmd z %{{
    let res = (zoxide query --exclude $env.PWD -- $@)
    lf -remote "send $env.id cd '$res'"
}}`

Unfortunately my directory is not changed when I type `z test`. 
If I run the first cmd in my nushell and then do `cd res`, I get to the right location. I have the feeling that the ID is not saved correctly. I also don't see an `id `variable under my env variables.

Does anyone have an idea or already done something with the nushell for zoxide?

Thanks in advance.
Matix

**Comment by mleby** [comment, marked as answer]
I use combination nushell, zoxide, alacritty, windows and this works for me:
```
cmd on-cd &{{
  zoxide add $env.PWD
}}

cmd zi ${{
  let result = (zoxide query -i)
  lf -remote  $"send ($env.id) cd \"($result | str replace --all '\' '\\')\""
}}
```
